source/ui/BioXAS/BioXASSIS3820ScalerView.cpp

```diff
@@ -0,0 +1,133 @@
+#include "BioXASSIS3820ScalerView.h"
+#include "ui/CLS/CLSSIS3820ScalerControlsView.h"
+#include "ui/BioXAS/BioXASSIS3820ScalerChannelsView.h"
+#include "ui/CLS/CLSSIS3820ScalerDarkCurrentWidget.h"
+
+BioXASSIS3820ScalerView::BioXASSIS3820ScalerView(CLSSIS3820Scaler *scaler, bool optionsShown, QWidget *parent) :
+    QWidget(parent)
+{
+	// Initialize member variables.
+
+	scaler_ = scaler;
+
+	// Create UI elements.
+
+	CLSSIS3820ScalerControlsView *controlsView = new CLSSIS3820ScalerControlsView(scaler_);
+
+	controlsBox_ = new QGroupBox();
+	controlsBox_->setTitle("Controls");
+
+	QCheckBox *biasEnabledShown = new QCheckBox("Bias voltage enabled");
+	QCheckBox *biasShown = new QCheckBox("Bias voltage");
+	QCheckBox *darkCurrentShown = new QCheckBox("Dark current");
+
+	optionsBox_ = new QGroupBox();
+	optionsBox_->setTitle("Options");
+
+	BioXASSIS3820ScalerChannelsView *channelsView = new BioXASSIS3820ScalerChannelsView(scaler_, false, false, false);
+
+	channelsBox_ = new QGroupBox();
+	channelsBox_->setTitle("Channels");
+
+	CLSSIS3820ScalerDarkCurrentWidget *darkCurrentWidget = new CLSSIS3820ScalerDarkCurrentWidget(scaler_);
+
+	darkCurrentBox_ = new QGroupBox();
+	darkCurrentBox_->setTitle("Dark current");
+
+	// Create and set layouts.
+
+	QHBoxLayout *controlsLayout = new QHBoxLayout();
+	controlsLayout->addStretch();
+	controlsLayout->addWidget(controlsView);
+	controlsLayout->addStretch();
+
+	controlsBox_->setLayout(controlsLayout);
+
+	QHBoxLayout *optionsLayout = new QHBoxLayout();
+	optionsLayout->addStretch();
+	optionsLayout->addWidget(biasEnabledShown);
+	optionsLayout->addWidget(biasShown);
+	optionsLayout->addWidget(darkCurrentShown);
+	optionsLayout->addStretch();
+
+	optionsBox_->setLayout(optionsLayout);
+
+	QVBoxLayout *channelsLayout = new QVBoxLayout();
+	channelsLayout->addWidget(channelsView);
+
+	channelsBox_->setLayout(channelsLayout);
+
+	QHBoxLayout *darkCurrentLayout = new QHBoxLayout();
+	darkCurrentLayout->addStretch();
+	darkCurrentLayout->addWidget(darkCurrentWidget);
+	darkCurrentLayout->addStretch();
+
+	darkCurrentBox_->setLayout(darkCurrentLayout);
+
+	QVBoxLayout *layout = new QVBoxLayout();
+	layout->addWidget(controlsBox_);
+	layout->addWidget(optionsBox_);
+	layout->addWidget(channelsBox_);
+	layout->addWidget(darkCurrentBox_);
+
+	setLayout(layout);
+
+	// Initial settings.
+
+	biasEnabledShown->setChecked(false);
+	biasShown->setChecked(false);
+	darkCurrentShown->setChecked(false);
+
+	optionsBox_->hide();
+	channelsBox_->hide();
+	darkCurrentBox_->hide();
+
+	// Make connections.
+
+	connect( biasEnabledShown, SIGNAL(clicked(bool)), channelsView, SLOT(setBiasEnabledEditorShown(bool)) );
```

Comment:
Just as a note, ```clicked(bool)``` is only the same as ```toggled(bool)``` when buttons are checkable.  You may want to be explicit about your calls, just in case.  Not a big deal though.